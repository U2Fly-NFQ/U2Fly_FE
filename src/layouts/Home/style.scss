@use '../../assets/scss/abstracts' as a;

.home-header {
  position: absolute;
  width: 100%;
  z-index: 99;
  padding-right: 0;
  padding-left: 0;
  background-color: transparent;
  line-height: normal;
}
