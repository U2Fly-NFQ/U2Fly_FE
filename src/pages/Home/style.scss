@use '../../assets/scss/abstracts/variables' as var;
@use '../../assets/scss/base/animation' as animate;

.animate__fadeInUp {
  animation: animate.$fadeInUp;
}

.animate__fadeInDown {
  animation: animate.$fadeInDown;
}

.animate__fadeInLeft {
  animation: animate.$fadeInLeft;
}

.animate__fadeInRight {
  animation: animate.$fadeInRight;
}

.btn {
  display: inline-block;
  font-weight: 400;
  line-height: 1.5;
  color: #212529;
  text-align: center;
  text-decoration: none;
  vertical-align: middle;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 1px solid transparent;
  padding: 0.375rem 0.75rem;
  font-size: 1rem;
  border-radius: 0.25rem;
  transition: color 0.15s;
  color: var.$white-color !important;

  &hover {
    margin-top: 2px;
  }
}

.btn_md {
  padding: 12px 35px;
  font-size: 18px;
}

.view-all-btn {
  display: grid;
}

.btn_theme {
  color: var.$white-color;
  background-color: var.$main-color;
  transition: var.$transition;
  box-shadow: none;
  overflow: hidden;
  white-space: nowrap;
  position: relative;
  z-index: 0;
  border: none;

  &:before {
    content: '';
    position: absolute;
    top: 0;
    bottom: 0;
    left: 50%;
    width: 580px;
    height: 550px;
    margin: auto;
    background: var(--ant-primary-5);
    border-radius: 50%;
    z-index: -1;
    -webkit-transform-origin: top center;
    transform-origin: top center;
    -webkit-transform: translateX(-50%) translateY(-5%) scale(0.4);
    transform: translateX(-50%) translateY(-5%) scale(0.4);
    transition: var.$transition;
  }

  &:hover::before {
    -webkit-transition: -webkit-transform 1s;
    transition: -webkit-transform 1s;
    transition: transform 1s;
    transition: transform 1s, -webkit-transform 1s;
    -webkit-transform: translateX(-45%) translateY(0) scale(1);
    transform: translateX(-50%) translateY(0) scale(1);
    -webkit-transform-origin: bottom center;
    transform-origin: bottom center;
    border: none;
  }
}

#go-beyond-area {
  margin-top: 100px;

  .heading_left_area__wrapper {
    display: flex;
    align-items: center;
    justify-content: center;
    height: 100%;
  }

  .heading_left_area {
    h2 {
      font-weight: 400;
      padding-bottom: 17px;
      line-height: 45px;
      font-size: 36px;

      span {
        display: inline-block;
        border-bottom: 1px solid var.$main-color;
        padding-bottom: 10px;
      }
    }

    h5 {
      color: var.$black-color-opacity;
      font-size: 18px;
      font-weight: 400;
    }
  }

  .imagination_boxed {
    position: relative;
    border-radius: 14px;
    overflow: hidden;
    cursor: pointer;

    & img {
      width: 100%;
      object-fit: cover;
      transition: var.$transition;
      height: 446px;
    }

    &:hover img {
      transform: scale(1.1);
    }

    &:hover::before {
      transform: translate(0%, 0%);
    }

    &::before {
      content: '';
      position: absolute;
      width: 100%;
      height: 25%;
      left: 0;
      background: #17161669;
      bottom: 0;
      z-index: 999;
      transition: all 0.5s;
      transform: translate(0%, 100%);
    }

    h3 {
      position: absolute;
      bottom: 0;
      display: flex;
      justify-content: center;
      width: 100%;
      z-index: 999;
      align-items: center;
      height: 25%;
      transition: var.$transition;
      font-size: 24px;
      flex-direction: column;
      color: var.$white-color;
      font-weight: 400;
    }

    &:hover h3 {
      font-weight: 600;
    }
  }
}

#top_destination {
  margin-top: 100px;

  .top_destination_heading {
    display: inline-bloc;
    position: relative;
    padding-bottom: 15px;
    font-size: 36px;
    font-weight: 500;
    line-height: 40px;
    margin-bottom: 30px;
    text-align: center;

    &::after {
      content: '';
      position: absolute;
      width: 200px;
      height: 1px;
      background: var.$main-color;
      left: 50%;
      transform: translate(-50%, 50%);
      bottom: 0;
    }
  }

  .destinations_content_box {
    position: relative;
    margin-bottom: 23px;
    border-radius: 12px;
    overflow: hidden;

    img {
      width: 100%;
    }

    .destinations_content_inner__left {
      @media screen and (max-width: 530px) {
        transform: scale(0.5) translate(-100%, -100%) !important;
      }
    }

    .destinations_content_inner {
      position: absolute;
      top: 50%;
      left: 50%;
      transform: translate(-50%, -50%);
      z-index: 99;

      a {
        color: var.$white-color;
        text-transform: capitalize;
        font-size: 24px;
      }

      h2 {
        color: var.$white-color;
        padding-bottom: 20px;
        font-size: 36px;
        font-weight: 500;
        line-height: 40px;
      }

      .destinations_big_offer {
        display: flex;

        h1 {
          color: var.$white-color;
          font-size: 150px;
          line-height: 109px;
          padding-bottom: 20px;
        }

        h6 {
          color: var.$white-color;
          line-height: 45px;
          padding-top: 8px;

          span {
            display: block;
            font-size: 38px;
            font-weight: 500;
          }
        }
      }
    }
  }
}
.img_animation {
  &:hover::after {
    opacity: 1;
    visibility: visible;
  }

  &:hover img {
    transform: scale(1.1);
  }

  &::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
    background: #8b3eea9c;
    left: 0;
    top: 0;
    opacity: 0;
    visibility: hidden;
    transition: var.$transition;
  }

  img {
    transition: var.$transition;
    position: relative;
  }
}

#offer-area {
  margin-top: 100px;

  .offer_area_box {
    position: relative;
    border-radius: 10px;
    overflow: hidden;

    img {
      width: 100%;
    }

    .offer_area_content {
      position: absolute;
      left: 0;
      bottom: 0;
      padding: 30px 27px;
      z-index: 999;

      h2 {
        color: var.$white-color;
        padding-bottom: 20px;
        font-size: 36px;
        font-weight: 500;
        line-height: 40px;
        @media screen and (max-width: 576px) {
          font-size: 24px;
        }

        @media screen and (max-width: 1100px) {
          font-size: 30px;
        }
      }

      p {
        color: var.$white-color;
        padding-bottom: 25px;
        font-size: 16px;
        line-height: 28px;
        font-weight: 400;
        font-family: 'Poppins', sans-serif;
        margin-bottom: 0;
        @media screen and (max-width: 576px) {
          display: none;
        }
      }
    }
  }
}
