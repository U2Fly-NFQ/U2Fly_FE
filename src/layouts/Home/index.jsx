import { Outlet } from 'react-router-dom'
import { Layout } from 'antd'
import './style.scss'

const { Header, Footer, Content } = Layout

function Home() {
  return (
    <Layout>
      <Layout>
        <Header>home header</Header>
        <Content>
          <Outlet />
        </Content>
        <Footer>home footer</Footer>
      </Layout>
    </Layout>
  )
}

export default Home
