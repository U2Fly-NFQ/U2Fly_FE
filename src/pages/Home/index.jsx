import './style.scss'
import { Row, Col } from 'antd'
import { Link } from 'react-router-dom'
import { HomeBanner } from '../../components'

function Home() {
  return (
    <div className="home-page">
      <HomeBanner />

      <div className="grid wide">
        <section id="go-beyond-area">
          <Row gutter={[24, 24]}>
            <Col span={24} sm={12} lg={6}>
              <div className="heading_left_area__wrapper">
                <div className="heading_left_area">
                  <h2>
                    Go beyond your <span>imagination</span>
                  </h2>
                  <h5>Discover your ideal experience with us</h5>
                </div>
              </div>
            </Col>
            <Col span={24} sm={12} lg={6}>
              <Link to="/flights">
                <div className="imagination_boxed">
                  <img
                    src="https://andit.co/projects/html/and-tour/assets/img/imagination/imagination1.png"
                    alt="img"
                  />

                  <h3>
                    7% Discount for all <span>Airlines</span>
                  </h3>
                </div>
              </Link>
            </Col>
            <Col span={24} sm={12} lg={6}>
              <Link to="/flights">
                <div className="imagination_boxed">
                  <img
                    src="https://andit.co/projects/html/and-tour/assets/img/imagination/imagination2.png"
                    alt="img"
                  />

                  <h3>
                    Travel around<span>the world</span>
                  </h3>
                </div>
              </Link>
            </Col>
            <Col span={24} sm={12} lg={6}>
              <Link to="/flights">
                <div className="imagination_boxed">
                  <img
                    src="https://andit.co/projects/html/and-tour/assets/img/imagination/imagination3.png"
                    alt="img"
                  />

                  <h3>
                    Luxury resorts<span>top deals</span>
                  </h3>
                </div>
              </Link>
            </Col>
          </Row>
        </section>

        <section id="top_destination">
          <h2 className="top_destination_heading">Top destinations</h2>

          <Row gutter={[24, 24]}>
            <Col span={24} lg={12}>
              <div className="destinations_content_box img_animation">
                <img
                  src="https://andit.co/projects/html/and-tour/assets/img/destination/big-img.png"
                  alt="img"
                />
                <div className="destinations_content_inner">
                  <h2>Up to</h2>
                  <div className="destinations_big_offer">
                    <h1>50</h1>
                    <h6>
                      <span>%</span> <span>Off</span>
                    </h6>
                  </div>
                  <h2>Holiday packages</h2>
                  <Link to="/flights" className="btn btn_theme btn_md">
                    Book now
                  </Link>
                </div>
              </div>
            </Col>
            <Col span={24} lg={12}>
              <Row gutter={24}>
                <Col span={24} sm={8}>
                  <div className="destinations_content_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/destination/destination1.png"
                      alt="img"
                    />
                    <div className="destinations_content_inner">
                      <Link to="/flights">China</Link>
                    </div>
                  </div>
                  <div className="destinations_content_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/destination/destination2.png"
                      alt="img"
                    />
                    <div className="destinations_content_inner">
                      <Link to="/flights">Darjeeling</Link>
                    </div>
                  </div>
                  <div className="destinations_content_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/destination/destination3.png"
                      alt="img"
                    />
                    <div className="destinations_content_inner">
                      <Link to="/flights">Malaysia</Link>
                    </div>
                  </div>
                </Col>
                <Col span={24} sm={8}>
                  <div className="destinations_content_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/destination/destination4.png"
                      alt="img"
                    />
                    <div className="destinations_content_inner">
                      <Link to="/flights">Gangtok</Link>
                    </div>
                  </div>
                  <div className="destinations_content_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/destination/destination5.png"
                      alt="img"
                    />
                    <div className="destinations_content_inner">
                      <Link to="/flights">Thailand</Link>
                    </div>
                  </div>
                  <div className="destinations_content_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/destination/destination6.png"
                      alt="img"
                    />
                    <div className="destinations_content_inner">
                      <Link to="/flights">Australia</Link>
                    </div>
                  </div>
                </Col>
                <Col span={24} sm={8}>
                  <div className="destinations_content_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/destination/destination7.png"
                      alt="img"
                    />
                    <div className="destinations_content_inner">
                      <Link to="/flights">London</Link>
                    </div>
                  </div>
                  <div className="destinations_content_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/destination/destination8.png"
                      alt="img"
                    />
                    <div className="destinations_content_inner">
                      <Link to="/flights">USA</Link>
                    </div>
                  </div>
                  <Link
                    to="/flights"
                    className="btn btn_theme btn_md view-all-btn"
                  >
                    View all
                  </Link>
                </Col>
              </Row>
            </Col>
          </Row>
        </section>

        <section id="offer-area">
          <Row gutter={[24, 24]}>
            <Col span={24} md={12}>
              <div className="offer_area_box img_animation">
                <img
                  src="https://andit.co/projects/html/and-tour/assets/img/offer/offer1.png"
                  alt="img"
                />
                <div class="offer_area_content">
                  <h2>Special Offers</h2>
                  <p>
                    Invidunt ut labore et dolore magna aliquyam erat, sed diam
                    voluptua. At vero eos et accusam et justo duo dolores et ea
                    rebum. Stet clita kasd dolor sit amet. Lorem ipsum dolor sit
                    amet.
                  </p>
                  <Link to="/flights" class="btn btn_theme btn_md">
                    Holiday deals
                  </Link>
                </div>
              </div>
            </Col>
            <Col span={24} md={12}>
              <Row gutter={[24, 24]}>
                <Col span={24} md={12}>
                  <div class="offer_area_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/offer/offer2.png"
                      alt="img"
                    />
                    <div class="offer_area_content">
                      <h2>News letter</h2>
                      <p>
                        Invidunt ut labore et dolore magna aliquyam erat, sed
                        diam voluptua. At vero eos et.
                      </p>
                      <Link to="/flights" class="btn btn_theme btn_md">
                        Subscribe now
                      </Link>
                    </div>
                  </div>
                </Col>
                <Col span={24} md={12}>
                  <div class="offer_area_box img_animation">
                    <img
                      src="https://andit.co/projects/html/and-tour/assets/img/offer/offer3.png"
                      alt="img"
                    />
                    <div class="offer_area_content">
                      <h2>Travel tips</h2>
                      <p>
                        Invidunt ut labore et dolore magna aliquyam erat, sed
                        diam voluptua. At vero eos et.
                      </p>
                      <Link to="/flights" class="btn btn_theme btn_md">
                        Get tips
                      </Link>
                    </div>
                  </div>
                </Col>
              </Row>
            </Col>
          </Row>
        </section>
      </div>
    </div>
  )
}

export default Home
