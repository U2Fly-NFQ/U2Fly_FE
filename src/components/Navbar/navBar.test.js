import renderer from 'react-test-renderer'
import React from 'react'
import { BrowserRouter as Router } from 'react-router-dom'
import Home from './'

describe('Navigation bar test', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Router>
          <Home />
        </Router>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
