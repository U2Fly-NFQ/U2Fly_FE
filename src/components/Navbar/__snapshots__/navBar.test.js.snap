// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation bar test renders correctly 1`] = `
<nav
  className="navbar"
>
  <div
    className="wide grid"
    style={
      Object {
        "justifyContent": "flex-end",
        "padding": "10px 0",
      }
    }
  >
    <ul
      className="navbar__links"
    >
      <div
        style={Object {}}
      >
        <h3
          className="ant-typography"
          onClick={[Function]}
          style={
            Object {
              "WebkitLineClamp": undefined,
              "alignItems": "center",
              "color": "#fff",
              "cursor": "pointer",
              "display": "flex",
              "fontSize": "22px",
              "gap": "0.6rem",
              "margin": 0,
              "marginRight": "20px",
            }
          }
        >
          U2FLy
        </h3>
      </div>
      <li
        className="navbar__link"
      >
        <a
          aria-current="page"
          className="nav-link active"
          href="/"
          onClick={[Function]}
        >
          <p
            className="navbar__link__data"
          >
            Home
             
          </p>
        </a>
      </li>
      <li
        className="navbar__link"
      >
        <a
          className="nav-link"
          href="/flights"
          onClick={[Function]}
        >
          <p
            className="navbar__link__data"
          >
            Flights
             
          </p>
        </a>
      </li>
    </ul>
    <div
      className="navbar__footer"
    >
      <div
        className="navbar__footer__btn"
      >
        <div
          className="form-submit"
        >
          <button
            className="btn btn-primary btn-md"
            id=""
            type="submit"
          >
            cta.become_partner
          </button>
        </div>
      </div>
    </div>
    <div
      className="navbar_icons"
      onClick={[Function]}
    >
      <div
        className="navbar_icon"
      />
    </div>
  </div>
  <div
    className="navbar__dropdown__modal"
  >
    <ul
      className="navbar__dropdown-menu"
    >
      <li
        className="navbar__dropdown-menu__item"
      >
        <a
          aria-current="page"
          className="active"
          href="/"
          onClick={[Function]}
        >
          header.navbar.home
        </a>
      </li>
      <li
        className="navbar__dropdown-menu__item"
      >
        <a
          className=""
          href="/flights"
          onClick={[Function]}
        >
          header.navbar.flight
        </a>
      </li>
    </ul>
  </div>
</nav>
`;
